brain/detail/circuit.h

```diff
@@ -398,7 +402,8 @@ class Circuit::Impl
 
     virtual GIDSet getGIDs(const std::string& target) const = 0;
 
-    GIDSet getRandomGIDs(const float fraction, const std::string& target) const
+    GIDSet getRandomGIDs(const float fraction, const std::string& target,
+                         const std::pair<bool, size_t> seed = {false, 0}) const
```

Comment:
Use size_t* size = nullptr for the optional seed parameter, that's a much more common idiom.